dataland-backend/backendOpenApi.json

```diff
@@ -453,7 +453,7 @@
           {
             "name": "dataTypes",
             "in": "query",
-            "required": false,
+            "required": true,
```

Comment:
Warum muss das ab jetzt required sein?
Bislang hat man ja quasi nach allen frameworks gesucht, wenn man keins angibt.
Wurde das in dieser story verworfen, und warum?

Reply:
Wir haben nun weitere Endpunkte, die wir benutzen können um Informationen über Companies zu erhalten. Wir setzen den Datatypes Endpunkt auf required, weil wir eh nur noch nichtleere Framework Filter akzeptieren. Der Grund dafür war... Query Geschwindigkeit? @stewal Was war da noch?

Reply:
@fsoffel Genau, das war der Grund, damit die Query nicht ewig braucht.

Reply:
Guter Punkt. Das würde sich ändern lassen. Wie kommunizieren wir aber klar, dass leer bedeutet, dass nach allen datatypes gesucht wird und nicht auch nach companies ohne irgendwelche Datensätze.

Reply:
Naja das war ja bislang auch so und bislang haben wir das auch nicht gesondert kommuniziert oder?

Ich stell mir das auch sehr aufwendig vor, wenn es in Zukunft z.B. 20 frameworks gibt, und du willst aber deine Suche gar nicht nach frameworks beschränken.

Für die anderen Filter bedeutet ja "leer" auch dass er nicht danach filtern soll, z.B. searchString oder sector oder country codes.
Da fände ich es konsistenter, wenn man dataTypes auch leer lassen dürfte.

Reply:
Stimmt eigentlich. Vielleicht sollten wir das aber in der Endpunktroute klarer machen, dass es nur Companies mit Frameworks gibt.

Reply:
Also ist unser Konsens:
- required wieder rückgängig machen aus Konsistenzgründen und Einfachheit für den User
- dafür aber klarer in der Doku erklären, dass nur companies mit mindestens einem dataset für mindestens ein framework returnt werden

Korrekt ?

Reply:
Meinst du nicht, wir sollen die Route anpassen?

Reply:
Was meinst du genau?

Reply:
Spreche ich im Refinement an